### DISCUSSION TITLE
EV3 to EV3 Communication

### DISCUSSION THREAD

**Discussion by shoringit** [pybricks/support#1406] [General]
I know that I can use Bluetooth communication between two EV3 bricks, but I'm wondering if there's a way to do it with a wired connection under pybricks.  I know that it doesn't support the USB daisy-chaining that was supported in EV3-G, but I'm wondering if it would be possible to communicate between them using I2C?...  Connect sensor-port to sensor-port and use IODevices to send/receive messages between the two programs...  Is this possible?  Has it been done before?...

Thanks,

Danny

**Comment by laurensvalk** [comment]
You could make a modified cable that swaps the two data wires, and then use `UARTDevice` on both EV3s.

**Reply by shoringit** [reply]
That's what I figured.  I may have to test this...

**Comment by dlech** [comment]
It's also possible to use TCP/IP sockets over the USB cable. 

With a little fiddling, the Pybricks mailbox stuff could be used with this (by replacing the Bluetooth RFCOMM sockets with regular network sockets).

**Reply by shoringit** [reply]
Good to know.  Thanks,
